src/visitors/kinetic_block_visitor.cpp

```diff
@@ -478,6 +478,162 @@ void KineticBlockVisitor::visit_kinetic_block(ast::KineticBlock& node) {
     kinetic_blocks.push_back(&node);
 }
 
+class LocalRateNames {
+    const std::string kf_stem = "kf";
+    const std::string kb_stem = "kb";
+    const std::string source_stem = "source";
+
+    std::shared_ptr<ast::Name> generate_local_name(const std::string& stem) {
+        auto name = std::make_shared<ast::Name>(
+            std::make_shared<ast::String>(fmt::format("{}{}_", stem, n_equations)));
+        local_names.push_back(name);
+
```

Comment:
just a general comment: one precaution we have to usually take is to see if the variable doesn't conflict with the existing variable. E.g. what is the chance that `kf_0` is an existing user-defined variable? 

One could check if the new variable already exists in the symbol table. IIRC, for a similar reason, methods like [suffix_random_string ](https://github.com/BlueBrain/nmodl/blob/735b11dba9a494a5139be2ad5b82ca15e12859db/src/visitors/visitor_utils.hpp#L45) were added. There is also `class SingletonRandomString`.

Reply:
Correct! Note, it's not `kf_0` but `kf0_` we need to dodge (with the hope that users have better taste than naming a variable with a trailing `_`).

The problem is readability I'd like to avoid `kf0_hweutwhr` and similar. It would be nicer if we could get access to `_kf0`; but it tells me that we can't have names that start with `_`. Which is good, if it would let me state that I'm not a user, and I can name my variables whatever I want.

Reply:
> with the hope that users have better taste than naming a variable with a trailing _

Understand! just that to me it's appears a bit stronger assumption if one assumes the user won't declare a variable with the name `X` :).

Just a suggestion: if you would like to keep readability aspect, see if it's easy just to do a check with `symtab->lookup_in_scope(name)`.

Reply:
We mangle as `kf0_{:04d}` and count upwards, checking both the symtab of the current statement block.